packages/vc-export/src/KiltAttestationProofV1.ts

```diff
@@ -0,0 +1,668 @@
+    Partial<KiltAttestationProofV1>
+): Uint8Array {
+  const { federatedTrustModel = [] } = credential
+  const { commitments } = proof
+  const rootHashInputs = [
+    // Collect commitments for root hash
+    ...commitments.map((i) => base58Decode(i)),
+    // Collect trust model items for root hash
+    ...federatedTrustModel.map((entry) => {
+      if (entry.type === KILT_ATTESTER_LEGITIMATION_V1_TYPE) {
+        // get root hash from credential id
+        return credentialIdToRootHash(entry.id)
+      }
+      if (entry.type === KILT_ATTESTER_DELEGATION_V1_TYPE) {
+        // get on-chain id from delegation id
+        return delegationIdFromAttesterDelegation(entry)
+      }
+      throw new CredentialMalformedError(
+        `unknown type ${
+          (entry as { type: string }).type
+        } in federatedTrustModel`
+      )
+    }),
+  ]
+  // Concatenate and hash to produce root hash
+  return blake2AsU8a(u8aConcatStrict(rootHashInputs), 256)
+}
+
+async function verifyAttestedAt(
+  claimHash: Uint8Array,
+  blockHash: Uint8Array,
+  opts: { api?: ApiPromise } = {}
+): Promise<{
+  verified: boolean
+  timestamp: number
+  attester: DidUri
+  cTypeId: ICType['$id']
+  delegationId: IDelegationNode['id'] | null
+}> {
+  const { api = ConfigService.get('api') } = opts
+  const apiAt = await api.at(blockHash)
+  // TODO: should we look up attestation storage as well and take attestation info from there? That gives us the definitive attestation state in this block (e.g. makes sure it hasn't been removed after)
+  const [events, time] = await apiAt.queryMulti<
+    [Vec<FrameSystemEventRecord>, u64]
+  >([
+    [api.query.system.events as QueryableStorageEntry<'promise'>],
+    [api.query.timestamp.now as QueryableStorageEntry<'promise'>],
+  ])
+
+  const timestamp = time.toNumber()
+  const attestationEvent = events
+    .reverse()
+    .find(
+      ({ phase, event }) =>
+        phase.isApplyExtrinsic &&
+        api.events.attestation.AttestationCreated.is(event) &&
+        u8aEq(event.data[1], claimHash)
+    )
+  if (!attestationEvent)
+    throw new SDKErrors.CredentialUnverifiableError(
+      `Matching attestation event for root hash ${u8aToHex(
+        claimHash
+      )} not found at block ${u8aToHex(blockHash)}`
+    )
+  const [att, , cTypeHash, authorization] = attestationEvent.event.data as [
+    AccountId,
+    Hash,
+    Hash,
+    Option<Hash> | Option<RuntimeCommonAuthorizationAuthorizationId>
+  ] &
+    IEventData
+  const attester = getFullDidUri(encodeAddress(att.toU8a(), 38))
+  const cTypeId = CType.hashToId(cTypeHash.toHex())
+  const delegationId = authorization.isSome
+    ? (
+        (authorization.unwrap() as RuntimeCommonAuthorizationAuthorizationId)
+          .value ?? authorization.unwrap()
+      ).toHex()
+    : null
+  return {
+    verified: true,
+    timestamp,
+    attester,
+    cTypeId,
+    delegationId,
+  }
+}
+
+async function findDelegators(
```

Comment:
Because it returns a `Promise<void>`, perhaps `find` is not the best term. What about `checkDelegation` or `checkDelegationHierarchy`?

Reply:
1b0f5389163fe4beec78c7c40de06a7b72f764f1